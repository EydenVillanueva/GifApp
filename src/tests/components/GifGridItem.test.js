import React from "react";
import {shallow} from "enzyme";
import GifGridItem from "../../components/GifGridItem";

describe('Test that component GifGridItem is showing correctly', () => {
	
	const title = "Test title";
	const url = "https://localhost/image.jpg"
	
	const wrapper = shallow(<GifGridItem url={url} title={title} />);
	
	test('It should correctly display the component ', () => {
			expect(wrapper).toMatchSnapshot();
	});
	
	test('It should have a paragraph with the title', () => {
		const p = wrapper.find('p');
		expect(p.text().trim()).toBe(title);
	});
	
	test('It should be the image equals than url props', () => {
		const img = wrapper.find('img');
		expect(img.props().src).toBe(url);
		expect(img.props().alt).toBe(title);
	});
});