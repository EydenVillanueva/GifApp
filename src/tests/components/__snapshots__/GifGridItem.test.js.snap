// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that component GifGridItem is showing correctly It should correctly display the component  1`] = `
<div
  className="card"
>
  <img
    alt="Test title"
    src="https://localhost/image.jpg"
  />
  <p>
    Test title
  </p>
</div>
`;
